components/playlist/browser/resources/media_detector/index.js

```diff
@@ -115,25 +113,18 @@
   }
 
   function getMediaDurationInSeconds (node) {
-    let duration = node.duration
-    const isNan = (value) => { return typeof value === 'number' && Number.isNaN(value) }
-    const isInfinite = (value) => { return typeof value === 'number' && (value === Infinity || value === -Infinity) }
     const clampDuration = (value) => {
-      if (isNan(value)) {
-        return 0.0
-      }
-
-      if (isInfinite(value)) {
-        return Number.MAX_VALUE
-      }
-      return value
+      if (Number.isFinite(value)) return value
+      if (value === Number.POSITIVE_INFINITY) return Number.MAX_VALUE
+      if (value === Number.NEGATIVE_INFINITY) return Number.MIN_VALUE
+      return 0.0
```

Comment:
Simple question so that I understand: when is a negative duration valid?

Reply:
Whoops, I had a pending review I hadn't submitted asking the same thing :D Sorry @sangwoo108 I thought I'd approved again

Reply:
Carried @fallaciousreasoning 's comment here together :)

> Just double checking that now we return -Infinity it's not going to break anything?

https://github.com/brave/brave-core/pull/16714/files#r1081758307

---

> when is a negative duration valid?

---

Yeah, good point. I was just translating the old code.

```
      if (isInfinite(value)) {
        return Number.MAX_VALUE
      }
```

I guess in case |value| is negative including infinite, returning 0 would make sense. What do you think about it?

Reply:
Yeah, that or leaving it as is is probably fine :smile:

Reply:
Thanks! changed to return 0 :)